Lib/cProfile.py

```diff
@@ -167,8 +167,12 @@ def main():
         else:
             progname = args[0]
             sys.path.insert(0, os.path.dirname(progname))
-            with open(progname, 'rb') as fp:
-                code = compile(fp.read(), progname, 'exec')
+            try:
+                with open(progname, 'rb') as fp:
+                    code = compile(fp.read(), progname, 'exec')
+            except FileNotFoundError:
```

Comment:
I would prefer to use a different CLI.

For example:

* file: ```python3 -m profile script.py```
* code: ```python3 -m profile -c "pass"```

Reply:
I also advocate for minimal changes the minimum complexity required to the code (btw I should write the test case)

What are the odds a filename can also be a valid python code ?

I thought about my great idea for 5 minutes and come to the following conclusion : 
Under POSIX rules possibility of collision DOES exists creating a little security concern.

I therefore finally agree.
My bad, I should have thought more about it.

I will see if I can write it. If I don't resubmit a patch in the next 48h it means I am chasing another rabbit that crossed my path

Reply:
> What are the odds a filename can also be a valid python code ?

Sure, you can have a filename called `pass`. Another example:

```
$ cat 1+1
print("Hello World")
$ ./python 1+1
Hello World
```